src/gfn/preprocessors.py

```diff
@@ -29,7 +29,11 @@ def preprocess(self, states: States) -> torch.Tensor:
     def __call__(self, states: States) -> torch.Tensor:
         """Transform the states to the input of the neural network, calling the preprocess method."""
         out = self.preprocess(states)
-        assert out.shape[-1] == self.output_dim
+        if isinstance(out, GeometricBatch):
+            assert out.x.shape[-1] == self.output_dim
```

Comment:
this is not needed, as you override `__call__` in `GraphPreprocessor`
As Graph is a special case, I would maybe remove this here